moveit_setup_assistant/moveit_setup_controllers/src/ros2_controllers_config.cpp

```diff
@@ -192,7 +192,7 @@ bool ROS2ControllersConfig::GeneratedControllersConfig::writeYaml(YAML::Emitter&
       emitter << YAML::Value;
       emitter << YAML::BeginMap;
       {
-        if (ci.joints_.size() != 1)
+        if (ci.type_ != "position_controllers/GripperActionController")
```

Comment:
While the diff will be slightly larger - maybe swap these statements so that we aren't working in negatives?
```
if (ci.type_ == "position_controllers/GripperActionController")
{
  emitter << YAML::Key << "joint" << YAML::Value << ci.joints_[0];
}
else
{
  emitter << YAML::Key << "joints" << YAML::Value << ci.joints_;
}
```

Reply:
The default behavior is to use the "joints" parameter as only the gripper controller uses the singular "joint". Either putting the "default" or "positive" clause first would work, but I think I'll leave it as is for now.